### PR TITLE
Add OS & shell information in 'info' command

### PR DESCRIPTION
**Pull request by thrijith**
#4599  add os and shell information in `wp cli info`

### CODE REVIEW

File: php/commands/src/CLI_Command.php [hunk 1]
```diff
@@ -103,6 +103,10 @@ public function info( $_, $assoc_args ) {
 
 		$runner = WP_CLI::get_runner();
 
+		$system_os           = php_uname( 's' );
```
Comment (gitlost) [inline php/commands/src/CLI_Command.php]: Could you move this block before `$runner = WP_CLI::get_runner();` please?

Also might as well give as much info as possible so:
```
$system_os = sprintf( '%s %s %s %s', php_uname( 's' ), php_uname( 'r' ), php_uname( 'v' ), php_uname( 'm' ) );
```
(only leaving out host name `'n'` which people mightn't want exposed)?

File: php/commands/src/CLI_Command.php [hunk 2]
```diff
@@ -103,6 +103,10 @@ public function info( $_, $assoc_args ) {
...
+		$system_architecture = php_uname( 'm' );
```
Comment (gitlost) [inline php/commands/src/CLI_Command.php]: So this `system_architecture` line would go.

File: php/commands/src/CLI_Command.php [hunk 3]
```diff
@@ -103,6 +103,10 @@ public function info( $_, $assoc_args ) {
...
+		$current_shell       = ( isset( $_SERVER['SHELL'] ) ) ? $_SERVER['SHELL'] : '';
```
Comment (gitlost) [inline php/commands/src/CLI_Command.php]: I think just `$shell` for the variable - "current" doesn't add anything. Also could you use `getenv()` instead and for Windows use `ComSpec` if `SHELL` not available:
```
$shell = getenv( 'SHELL' );
if ( ! $shell && Utils\is_windows() ) {
	$shell = getenv( 'ComSpec' );
}
```

File: php/commands/src/CLI_Command.php [hunk 4]
```diff
@@ -115,6 +119,9 @@ public function info( $_, $assoc_args ) {
 				'wp_cli_dir_path' => WP_CLI_ROOT,
 				'wp_cli_packages_dir_path' => $packages_dir,
 				'wp_cli_version' => WP_CLI_VERSION,
+				'system_os' => $system_os,
+				'system_architecture' => $system_architecture,
```
Comment (gitlost) [inline php/commands/src/CLI_Command.php]: So this line `system_architecture` goes.

File: php/commands/src/CLI_Command.php [hunk 5]
```diff
@@ -115,6 +119,9 @@ public function info( $_, $assoc_args ) {
...
+				'current_shell' => $current_shell,
```
Comment (gitlost) [inline php/commands/src/CLI_Command.php]: And this becomes:
```
'shell' => $shell,
```

File: php/commands/src/CLI_Command.php [hunk 6]
```diff
@@ -129,6 +136,9 @@ public function info( $_, $assoc_args ) {
 			WP_CLI::line( "WP-CLI global config:\t" . $runner->global_config_path );
 			WP_CLI::line( "WP-CLI project config:\t" . $runner->project_config_path );
 			WP_CLI::line( "WP-CLI version:\t" . WP_CLI_VERSION );
+			WP_CLI::line( "Operating System:\t" . $system_os );
```
Comment (gitlost) [inline php/commands/src/CLI_Command.php]: Thinking just shorten `Operating System` to `OS`.

File: php/commands/src/CLI_Command.php [hunk 7]
```diff
@@ -129,6 +136,9 @@ public function info( $_, $assoc_args ) {
...
+			WP_CLI::line( "System Architecture:\t" . $system_architecture );
```
Comment (gitlost) [inline php/commands/src/CLI_Command.php]: And this line `system_architecture` goes.

File: php/commands/src/CLI_Command.php [hunk 8]
```diff
@@ -129,6 +136,9 @@ public function info( $_, $assoc_args ) {
...
+			WP_CLI::line( "Current Shell:\t" . $current_shell );
```
Comment (gitlost) [inline php/commands/src/CLI_Command.php]: And this line becomes:
```
WP_CLI::line( "Shell:\t" . $shell );
```
Also could you move both lines to the start, before `PHP binary`? Seems more logical to me and looks slightly better...

(The formatting here with tabs isn't good and we should be using a table but we'll leave it for the moment...)

File: php/commands/src/CLI_Command.php (hunk 1 above)
Comment (thrijith) [reply php/commands/src/CLI_Command.php]: ok @gitlost  on it

File: php/commands/src/CLI_Command.php (hunk 3 above)
Comment (thrijith) [reply php/commands/src/CLI_Command.php]: thanks @gitlost for this
```
if ( ! $shell && Utils\is_windows() ) {
	$shell = getenv( 'ComSpec' );
}
```
i didn't know about this.

Comment (gitlost) [reply php/commands/src/CLI_Command.php]: Yeah, I've become a bit of a Windows guru over the last few weeks (don't use it locally or for servers)! `ComSpec` always returns path to `cmd.exe`, even on PowerShell, so isn't that great but looks a bit better anyway...

Comment (thrijith) [reply php/commands/src/CLI_Command.php]: I tested the code on Windows 7 ( installed VirtualBox just for that ) works perfectly:100:

Comment (gitlost) [reply php/commands/src/CLI_Command.php]: Cool, thanks @thrijith !

**Pull request merged**